toolbox/src/RelativeTransform.cpp

```diff
@@ -0,0 +1,246 @@
+    ok = ok && blockInfo->setOutputPortType(OUTPUT_IDX_TRANSFORM, DataType::DOUBLE);
+
+    if (!ok) {
+        wbtError << "Failed to configure output ports.";
+        return false;
+    }
+
+    return true;
+}
+
+bool RelativeTransform::initialize(BlockInformation* blockInfo)
+{
+    if (!WBBlock::initialize(blockInfo)) {
+        return false;
+    }
+
+    // INPUT PARAMETERS
+    // ================
+
+    if (!parseParameters(blockInfo)) {
+        wbtError << "Failed to parse parameters.";
+        return false;
+    }
+
+    std::string frame1;
+    std::string frame2;
+
+    bool ok = true;
+    ok = ok && m_parameters.getParameter("Frame1", frame1);
+    ok = ok && m_parameters.getParameter("Frame2", frame2);
+
+    if (!ok) {
+        wbtError << "Failed to get parameters after their parsing.";
+        return false;
+    }
+
+    // Check if the frames are valid
+    // -----------------------------
+
+    const auto kinDyn = getKinDynComputations(blockInfo).lock();
+    if (!kinDyn) {
+        wbtError << "Cannot retrieve handle to KinDynComputations.";
+        return false;
+    }
+
+    // Frame 1
+    pImpl->frame1Index = kinDyn->getFrameIndex(frame1);
+    if (pImpl->frame1Index == iDynTree::FRAME_INVALID_INDEX) {
+        wbtError << "Cannot find " + frame1 + " in the frame list.";
+        return false;
+    }
+
+    // Frame 2
+    pImpl->frame2Index = kinDyn->getFrameIndex(frame2);
+    if (pImpl->frame2Index == iDynTree::FRAME_INVALID_INDEX) {
+        wbtError << "Cannot find " + frame2 + " in the frame list.";
+        return false;
+    }
+
+    return true;
+}
+
+bool RelativeTransform::terminate(const BlockInformation* blockInfo)
+{
+    return WBBlock::terminate(blockInfo);
+}
+
+bool RelativeTransform::output(const BlockInformation* blockInfo)
+{
+    using namespace Eigen;
+    using Matrix4dSimulink = Matrix<double, 4, 4, Eigen::ColMajor>;
+    using Matrix4diDynTree = Matrix<double, 4, 4, Eigen::RowMajor>;
+
+    // Get the KinDynComputations object
+    auto kinDyn = getKinDynComputations(blockInfo).lock();
+    if (!kinDyn) {
+        wbtError << "Failed to retrieve the KinDynComputations object.";
+        return false;
+    }
+
+    // GET THE SIGNALS POPULATE THE ROBOT STATE
+    // ========================================
+
+    const Signal jointsPosSig = blockInfo->getInputPortSignal(INPUT_IDX_JOINTCONF);
+
+    if (!jointsPosSig.isValid()) {
+        wbtError << "Input signals not valid.";
+        return false;
+    }
+
+    // Create a Signal with an identity base
+    auto fakeBase = Signal(Signal::DataFormat::CONTIGUOUS,
+                           DataType::DOUBLE,
+                           /*isConst=*/false);
+    fakeBase.setWidth(16);
+    fakeBase.initializeBufferFromContiguous(
+        iDynTree::Transform::Identity().asHomogeneousTransform().data());
+
+    bool ok = setRobotState(&fakeBase, &jointsPosSig, nullptr, nullptr, kinDyn.get());
```

Comment:
You can also use [``setJointPos``](http://wiki.icub.org/codyco/dox/html/idyntree/html/classiDynTree_1_1KinDynComputations.html#a55c2232d8d0069af6a3e80d0f901be02).

Reply:
Despite the name is the same as `KinDynComputations::setRobotState`, it is not :sweat_smile: This method is [`WBBlock::setRobotState`](https://robotology.github.io/wb-toolbox/devel/doxygen/classwbt_1_1_w_b_block.html#a990354bac76a31d3baba9cf92c4b96cd).

Reply:
Ah dammit I did not notice it. Eventually you could wrap also that one :grin:

Reply:
Just for curiosity, do you have to configure somehow the `world_H_base` transform in order to use `setJointPos`? From the documentation is written that it uses the previous value, but what if it has never been set?

cc @traversaro

Reply:
I think the initial value is always identity, but I am not sure this is documented.